pkg/kv/kvserver/rangefeed/stream_manager.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package rangefeed
+
+import (
+	"context"
+	"sync"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+)
+
+// StreamManager manages one or more streams.
+//
+// Implemented by BufferedSender and StreamManager.
+type StreamManager struct {
+	// taskCancel is a function to cancel UnbufferedSender.run spawned in the
+	// background. It is called by UnbufferedSender.Stop. It is expected to be
+	// called after UnbufferedSender.Start.
+	taskCancel context.CancelFunc
+
+	// wg is used to coordinate async tasks spawned by UnbufferedSender.
+	// Currently, there is only one task spawned by UnbufferedSender.Start
+	// (UnbufferedSender.run).
+	wg sync.WaitGroup
+
+	// errCh is used to signal errors from UnbufferedSender.run back to the
+	// caller. If non-empty, the UnbufferedSender.run is finished and error should
+	// be handled. Note that it is possible for UnbufferedSender.run to be
+	// finished without sending an error to errCh. Other goroutines are expected
+	// to receive the same shutdown signal in this case and handle error
+	// appropriately.
```

Comment:
I removed it since it could be confusing, but the intent was to justify why we don’t return errors when the context is cancelled or when the stopper is quiescing. node.MuxRangefeed should be shutting down since it would be receiving the context cancellation and stopper quiescing.